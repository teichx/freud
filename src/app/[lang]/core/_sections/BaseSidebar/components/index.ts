export * from './ButtonToggleSidebar';
export * from './SidebarButton';
export * from './SidebarWrapper';
