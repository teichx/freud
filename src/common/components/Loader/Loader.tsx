import { FC } from 'react';

import { CircularProgress } from '@chakra-ui/react';

import { StyledLoader } from './style';
import { LoaderProps } from './types';

export const Loader: FC<LoaderProps> = ({
  isLoading,
  color = 'book.desertSun.500',
}) => (
  <StyledLoader data-is-loading={isLoading}>
    <CircularProgress isIndeterminate color={color} />
  </StyledLoader>
);
