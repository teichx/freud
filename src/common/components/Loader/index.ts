export * from './Loader';
export * from './style';
export * from './types';
