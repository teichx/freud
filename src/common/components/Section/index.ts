export * from './Section';
export * from './style';
export * from './types';
