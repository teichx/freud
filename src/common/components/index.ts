export * from './Avatar';
export * from './Breadcrumb';
export * from './Buttons';
export * from './DataModal';
export * from './DataModal';
export * from './DataTable';
export * from './Display';
export * from './Form';
export * from './Input';
export * from './GoogleLogin';
export * from './InfiniteScroll';
export * from './Loader';
export * from './Logo';
export * from './Meta';
export * from './ReadOnlyText';
export * from './Section';
export * from './Select';
export * from './SkeletonContextLoader';
export * from './TextHighlight';
export * from './ToggleTheme';
export * from './TooltipComponent';
export * from './TooltipDate';
