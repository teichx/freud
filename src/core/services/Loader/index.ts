export * from './LoaderConstants';
export * from './LoaderReducer';
export * from './types';
export * from './useLoader';
