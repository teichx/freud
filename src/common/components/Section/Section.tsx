import { FC, PropsWithChildren } from 'react';

import { useLoader } from '~/core/services';

import { Loader } from '../Loader/Loader';
import { ChildrenWrapper, SectionText, SectionWrapper } from './style';
import { SectionProps } from './types';

export const Section: FC<PropsWithChildren<SectionProps>> = ({
  label,
  children,
  disabledLoading,
  ...props
}) => {
  const { isLoading } = useLoader();

  return (
    <SectionWrapper {...props} as='fieldset'>
      {!label ? null : (
        <SectionText as='legend' variant='xl'>
          {label}
        </SectionText>
      )}

      <ChildrenWrapper p='2'>{children}</ChildrenWrapper>

      <Loader isLoading={isLoading && !disabledLoading} />
    </SectionWrapper>
  );
};
