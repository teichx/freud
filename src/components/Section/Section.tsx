import { FC, PropsWithChildren } from 'react';

import { CircularProgress } from '@chakra-ui/react';

import { useLoader } from '~/services/Loader';

import {
  ChildrenWrapper,
  SectionLoader,
  SectionText,
  SectionWrapper,
} from './style';
import { SectionProps } from './types';

export const Section: FC<PropsWithChildren<SectionProps>> = ({
  label,
  children,
  disabledLoading,
  ...props
}) => {
  const { isLoading } = useLoader();

  return (
    <SectionWrapper {...props} as='fieldset'>
      {!label ? null : (
        <SectionText as='legend' variant='xl'>
          {label}
        </SectionText>
      )}

      <ChildrenWrapper p='2'>{children}</ChildrenWrapper>

      <SectionLoader data-is-loading={isLoading && !disabledLoading}>
        <CircularProgress isIndeterminate color='book.desertSun.500' />
      </SectionLoader>
    </SectionWrapper>
  );
};
