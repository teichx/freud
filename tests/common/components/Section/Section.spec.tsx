jest.mock('~/core/services/Loader/useLoader');
import { render } from 'test-utils';

import { Section } from '~/common/components/Section';
import * as services from '~/core/services/Loader/useLoader';

describe('component.Section', () => {
  beforeEach(() => {
    jest.spyOn(services, 'useLoader').mockReturnValue({
      isLoading: false,
      setIsLoading: jest.fn(),
      startLoading: jest.fn(),
      endLoading: jest.fn(),
    });
  });

  it('renders without label', () => {
    expect.assertions(1);

    const { asFragment } = render(<Section>one children</Section>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders not loading', () => {
    expect.assertions(1);

    const { asFragment } = render(
      <Section label='<label>'>one children</Section>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders when loading', () => {
    expect.assertions(1);

    jest.spyOn(services, 'useLoader').mockReturnValue({
      isLoading: true,
      setIsLoading: jest.fn(),
      startLoading: jest.fn(),
      endLoading: jest.fn(),
    });

    const { asFragment } = render(
      <Section label='<label>'>one children</Section>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders when loading having disabled loading', () => {
    expect.assertions(1);

    jest.spyOn(services, 'useLoader').mockReturnValue({
      isLoading: true,
      setIsLoading: jest.fn(),
      startLoading: jest.fn(),
      endLoading: jest.fn(),
    });

    const { asFragment } = render(
      <Section label='<label>' disabledLoading>
        one children
      </Section>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('different when has disabledLoading', () => {
    expect.assertions(1);

    jest.spyOn(services, 'useLoader').mockReturnValue({
      isLoading: true,
      setIsLoading: jest.fn(),
      startLoading: jest.fn(),
      endLoading: jest.fn(),
    });

    const { asFragment: fragmentDisabledLoading } = render(
      <Section label='<label>' disabledLoading>
        one children
      </Section>
    );

    const { asFragment: fragmentEnabledLoading } = render(
      <Section label='<label>'>one children</Section>
    );

    expect(fragmentDisabledLoading()).not.toStrictEqual(
      fragmentEnabledLoading()
    );
  });
});
