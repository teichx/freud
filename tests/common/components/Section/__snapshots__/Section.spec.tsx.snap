// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.Section renders not loading 1`] = `
<DocumentFragment>
  <fieldset
    class="css-189pxlq"
  >
    <legend
      class="css-uk26kh"
      variant="xl"
    >
      &lt;label&gt;
    </legend>
    <div
      class="css-187wvck"
    >
      one children
    </div>
    <div
      class="css-19tk458"
      data-is-loading="false"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        class="chakra-progress css-120wkjd"
        data-indeterminate=""
        role="progressbar"
      >
        <svg
          class="css-jf70f3"
          viewBox="0 0 100 100"
        >
          <circle
            class="chakra-progress__track css-ol3i12"
            cx="50"
            cy="50"
            r="42"
            stroke-width="10px"
          />
          <circle
            class="chakra-progress__indicator css-1dsne73"
            cx="50"
            cy="50"
            r="42"
            stroke-width="10px"
          />
        </svg>
      </div>
    </div>
  </fieldset>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`component.Section renders when loading 1`] = `
<DocumentFragment>
  <fieldset
    class="css-189pxlq"
  >
    <legend
      class="css-uk26kh"
      variant="xl"
    >
      &lt;label&gt;
    </legend>
    <div
      class="css-187wvck"
    >
      one children
    </div>
    <div
      class="css-17rq0zv"
      data-is-loading="true"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        class="chakra-progress css-120wkjd"
        data-indeterminate=""
        role="progressbar"
      >
        <svg
          class="css-jf70f3"
          viewBox="0 0 100 100"
        >
          <circle
            class="chakra-progress__track css-ol3i12"
            cx="50"
            cy="50"
            r="42"
            stroke-width="10px"
          />
          <circle
            class="chakra-progress__indicator css-1dsne73"
            cx="50"
            cy="50"
            r="42"
            stroke-width="10px"
          />
        </svg>
      </div>
    </div>
  </fieldset>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`component.Section renders when loading having disabled loading 1`] = `
<DocumentFragment>
  <fieldset
    class="css-189pxlq"
  >
    <legend
      class="css-uk26kh"
      variant="xl"
    >
      &lt;label&gt;
    </legend>
    <div
      class="css-187wvck"
    >
      one children
    </div>
    <div
      class="css-19tk458"
      data-is-loading="false"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        class="chakra-progress css-120wkjd"
        data-indeterminate=""
        role="progressbar"
      >
        <svg
          class="css-jf70f3"
          viewBox="0 0 100 100"
        >
          <circle
            class="chakra-progress__track css-ol3i12"
            cx="50"
            cy="50"
            r="42"
            stroke-width="10px"
          />
          <circle
            class="chakra-progress__indicator css-1dsne73"
            cx="50"
            cy="50"
            r="42"
            stroke-width="10px"
          />
        </svg>
      </div>
    </div>
  </fieldset>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`component.Section renders without label 1`] = `
<DocumentFragment>
  <fieldset
    class="css-189pxlq"
  >
    <div
      class="css-187wvck"
    >
      one children
    </div>
    <div
      class="css-19tk458"
      data-is-loading="false"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        class="chakra-progress css-120wkjd"
        data-indeterminate=""
        role="progressbar"
      >
        <svg
          class="css-jf70f3"
          viewBox="0 0 100 100"
        >
          <circle
            class="chakra-progress__track css-ol3i12"
            cx="50"
            cy="50"
            r="42"
            stroke-width="10px"
          />
          <circle
            class="chakra-progress__indicator css-1dsne73"
            cx="50"
            cy="50"
            r="42"
            stroke-width="10px"
          />
        </svg>
      </div>
    </div>
  </fieldset>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
